### ISSUE TITLE
The sparse4dv2 model fails to converge

### ISSUE THREAD

**Issue by wuminghu666** [linxuewu/Sparse4D#34]
I trained the model using the unmodified code and data, expecting to reproduce sparse4d. However, sparse4dv2 failed to converge, and the grad_norm encountered NaN or Infinity, resulting in abnormal results. Do you have any suggestions on how to resolve this issue? Additionally, when can we expect the release of the Sparse4D-v3 code? I'm eagerly awaiting your response. Thank you!

**Comment by linxuewu** [comment]
Please check the package version first, especially whether the version of mmdet3d is greater than 1.0

**Comment by wuminghu666** [comment]
> Please check the package version first, especially whether the version of mmdet3d is greater than 1.0

Thank you so much. I gave it another try, and the issue has been successfully resolved!

**Comment by linxuewu** [comment]
> > Please check the package version first, especially whether the version of mmdet3d is greater than 1.0
> 
> Thank you so much. I gave it another try, and the issue has been successfully resolved!

That's great. I will publish our paper first and then make the code open source as soon as possible. Due to busy work, the code may need to wait for a while.

**Comment by WangYichen222** [comment]
I encountered the same error. Is it related to the version of mmdet3d? The version I am currently using is 1.0.0rc4.

**Comment by linxuewu** [comment]
Please check the definition of yaw=0 is same as the x axis of lidar coordinate in `mmdet3d/core/bbox/structures/lidar_box3d.py`.

If they are the same, it is not caused by this problem.

**Comment by sidiangongyuan** [comment]
> > I encountered the same error. Is it related to the version of mmdet3d? The version I am currently using is 1.0.0rc4.
> 
> Please check the definition of yaw=0 is same as the x axis of lidar coordinate in `mmdet3d/core/bbox/structures/lidar_box3d.py`.
> 
> If they are the same, it is not caused by this problem.

I encountered the same error in train step. And my env is the same as the requirement you provided.   
What's mean "Please check the definition of yaw=0."   Can you provide more info?

**Comment by linxuewu** [comment]
The direction of yaw=0 needs to be consistent with the x-axis.
![image](https://github.com/linxuewu/Sparse4D/assets/50012935/56665d54-676d-408d-86f5-b645ae4669c9)


> > > I encountered the same error. Is it related to the version of mmdet3d? The version I am currently using is 1.0.0rc4.
> > 
> > 
> > Please check the definition of yaw=0 is same as the x axis of lidar coordinate in `mmdet3d/core/bbox/structures/lidar_box3d.py`.
> > If they are the same, it is not caused by this problem.
> 
> I encountered the same error in train step. And my env is the same as the requirement you provided. What's mean "Please check the definition of yaw=0." Can you provide more info?

**Comment by sidiangongyuan** [comment]
I checked my LiDARInstance3DBoxes; they're the same as you mentioned. 
Here is my training log.  

<img width="623" alt="Screenshot 2023-12-13 085922" src="https://github.com/linxuewu/Sparse4D/assets/74222548/38456834-3caa-4bb6-b6b2-96855a718634">

It seems difficult to converge and the results on val are poor.
img_bbox_NuScenes/NDS: 0.0279, img_bbox_NuScenes/mAP: 0.0092

**Comment by linxuewu** [comment]
> I checked my LiDARInstance3DBoxes; they're the same as you mentioned. Here is my training log.
> 
> <img alt="Screenshot 2023-12-13 085922" width="623" src="https://private-user-images.githubusercontent.com/74222548/290042618-38456834-3caa-4bb6-b6b2-96855a718634.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1Mjc4NDgsIm5iZiI6MTcwMjUyNzU0OCwicGF0aCI6Ii83NDIyMjU0OC8yOTAwNDI2MTgtMzg0NTY4MzQtM2NhYS00YmI2LWI2YjItOTY4NTVhNzE4NjM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDA0MTkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxYTdkM2M5NDE2OWY4OWRmODhkNDBhYTQzYjNlMjFhMDE3NDA2ODhhNTA0NDY3NTUyOWJlNjk1NDNjOTY0YTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NFSAraVrkOWTLhovLC5lqSGTOUTnOag9emLcN6vxXic">
> It seems difficult to converge and the results on val are poor. img_bbox_NuScenes/NDS: 0.0279, img_bbox_NuScenes/mAP: 0.0092
> 
> I use the config : sparse4dv2_r50_HInf_256x704.py, I don't use pretrained resnet50.

Pre-trained weights are necessary. Try adding pre-training?

**Comment by sidiangongyuan** [comment]
> > I checked my LiDARInstance3DBoxes; they're the same as you mentioned. Here is my training log.
> > <img alt="Screenshot 2023-12-13 085922" width="623" src="https://private-user-images.githubusercontent.com/74222548/290042618-38456834-3caa-4bb6-b6b2-96855a718634.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDI1Mjc4NDgsIm5iZiI6MTcwMjUyNzU0OCwicGF0aCI6Ii83NDIyMjU0OC8yOTAwNDI2MTgtMzg0NTY4MzQtM2NhYS00YmI2LWI2YjItOTY4NTVhNzE4NjM0LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE0VDA0MTkwOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWMxYTdkM2M5NDE2OWY4OWRmODhkNDBhYTQzYjNlMjFhMDE3NDA2ODhhNTA0NDY3NTUyOWJlNjk1NDNjOTY0YTYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.NFSAraVrkOWTLhovLC5lqSGTOUTnOag9emLcN6vxXic">
> > It seems difficult to converge and the results on val are poor. img_bbox_NuScenes/NDS: 0.0279, img_bbox_NuScenes/mAP: 0.0092
> > I use the config : sparse4dv2_r50_HInf_256x704.py, I don't use pretrained resnet50.
> 
> Pre-trained weights are necessary. Try adding pre-training?

ok, I will try.  I use pretrained ResNet101 you provide, it looks normal.

**Comment by feimadada** [comment]
hello, mmdet3d 1.0.0rc6的版本是ok的吗？
import mmdet3d报错是什么问题？
![image](https://github.com/user-attachments/assets/8f388766-3b10-4815-9c97-7c9f20578404)

**Comment by sidiangongyuan** [comment]
mmcv-full                 1.4.0                    pypi_0    pypi
mmdet                     2.19.1                   pypi_0    pypi
mmdet3d                   1.0.0rc0                  dev_0    <develop>
mmsegmentation            0.20.2                   pypi_0    pypi
这是我的版本。 可以正常运行
